### PR TITLE
Add PackageSigning module

### PR DESCRIPTION
**Pull request by yim-lee**
Motivation:
Define API for package signing and signature validation in order to make progress on registry publication.

Changes:
- New `PackageSigning` module
- Add API stubs for package signature
- Add model stubs related to package signing

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please test

File: Sources/PackageSigning/Signing.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    public init() {
+    }
+
+    public func sign(
+        _ content: Data,
+        with identity: SigningIdentity,
+        in format: SignatureFormat,
+        observabilityScope: ObservabilityScope,
+        callbackQueue: DispatchQueue,
+        completion: @escaping (Result<Data, Error>) -> Void
```
Comment (yim-lee) [inline Sources/PackageSigning/Signing.swift:28]: Do we want callback or async/await?

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageSigning/Signing.swift (hunk 1 above)
Comment (neonichu) [reply Sources/PackageSigning/Signing.swift:28]: Don't have strong opinions, but if we want/need this to work on macOS CI, we cannot use async/await.

Comment (tomerd) [reply Sources/PackageSigning/Signing.swift:28]: starting to move things to async/await is better. the CI issues will wash away sooner or later

Comment (tomerd) [reply Sources/PackageSigning/Signing.swift:28]: cc @shahmishal for the CI issues

Comment (neonichu) [reply Sources/PackageSigning/Signing.swift:28]: Hard for me to tell, we may actually need newer macOS. It depends on whether Xcode offering the concurrency backwards compatibility library will be sufficient or if it has to come from the superior toolchain.

Comment (yim-lee) [reply Sources/PackageSigning/Signing.swift]: 👍 changing to async/await

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test Windows

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test Windows

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test Windows

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test Windows

**Comment by yim-lee** [comment]
@swift-ci please test Windows

**Review by neonichu — APPROVED** [review]

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test Windows

**Pull request merged**